public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/route-settings/MuteTimingFields.tsx

```diff
@@ -44,3 +46,21 @@ export function MuteTimingFields({ alertManager }: MuteTimingFieldsProps) {
     </Field>
   );
 }
+
+function getMuteTimings(): Array<SelectableValue<string>> {
```

Comment:
This function is pretty tricky - it's a hook (due to useQuery) and its asynchronous (because we need to wait for a request) but here we pretend that it's just a simple function
I believe it should be changed to a hook, like `useMuteTimings)

Additionally I think it would be better to return plain mute timings and do the conversion to `SelectableValue` as late as possible. That approach would improve reusability of the hook

Reply:
> I think it would be better to return plain mute timings and do the conversion to SelectableValue as late as possible. That approach would improve reusability of the hook

good idea

Reply:
> Additionally I think it would be better to return plain mute timings and do the conversion to SelectableValue as late as possible. That approach would improve reusability of the hook

Re thinking about this... I think for reusing the hook, we already have the generic `alertmanagerApi.endpoints.getMuteTimingList.useQuery` hook for this purpose. 
This hook in here is very specific, to transform the result of the generic hook and also to decide the `refetchOnFocus`, and `refetchOnReconnect` options.

Reply:
So in that case I'd consider changing the name of it to reflect that specificity. Right now, the name is very generic but the code is specific.
Maybe something like `useSelectableMuteTimings` would be better

Reply:
done!